midtransclient/http_client.py

```diff
@@ -39,14 +39,30 @@ def request(self, method, server_key, request_url, parameters=dict(),
         default_headers = {
             'content-type': 'application/json',
             'accept': 'application/json',
-            'user-agent': 'midtransclient-python/1.2.0'
+            'user-agent': 'midtransclient-python/1.5.0'
         }
         headers = default_headers
 
         # only merge if custom headers exist
         if custom_headers:
             headers = {**default_headers, **custom_headers}
 
+        # raise Midtrans error if server_key is not valid format
+        if server_key == "" or server_key is None:
+            raise MidtransAPIError(
+                message='The ServerKey is invalid, as it is an empty string or Null. Please double-check your API key. You can check the ServerKey from the Midtrans Dashboard. See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys for the details or contact support at support@midtrans.com if you have any questions.',
+                api_response_dict=None,
+                http_status_code=0,
+                raw_http_client_data=None
+            )
+        if " " in server_key:
+            raise MidtransAPIError(
+                message='The ServerKey is contains white-space. Please double-check your API key. You can check the ServerKey from the Midtrans Dashboard. See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys for the details or contact support at support@midtrans.com if you have any questions.',
+                api_response_dict=None,
+                http_status_code=0,
+                raw_http_client_data=None
+            )
+
```

Comment:
Grammar issue, no need `is`.


Btw this is my opinion, but feel free to decide. I don't think we should check for whitespace or any specific chars, because the Spec can change from API side. And if it changed, we will need to edit this kind of logic manually each time. Also I think the API raw error message in this scenario already tell merchant that ServerKey is invalid.

So IMO the benefit is small & the potential maintenance burden is higher.

Reply:
Sure will remove this part. Thanks da